dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcContext.java

```diff
@@ -731,12 +731,11 @@ public void asyncCall(Runnable runnable) {
     @SuppressWarnings("unchecked")
     public static AsyncContext startAsync() throws IllegalStateException {
         RpcContext currentContext = getContext();
-        if (currentContext.asyncContext != null) {
-            currentContext.asyncContext.start();
-            return currentContext.asyncContext;
-        } else {
-            throw new IllegalStateException("This service does not support asynchronous operations, you should open async explicitly before use.");
+        if (currentContext.asyncContext == null) {
+            currentContext.asyncContext = new AsyncContextImpl();

```

Comment:
As here we are setting asyncContext  if it is null and then starting. Where we have a separate public method to set context asyncContext. In the scenario if we started the context after creating an instance of AsyncContextImpl. Down the line if caller modify sets the asyncContext by **setAsyncContext** method of this class, then we will not be having any track of lost instance and we will not be able to release resource or store the context. 

What do you say?

Reply:
We should not allow user to set asynccontext(), like async servlet, we should only allow user to call 'startAsync()' to get a async context. 

I think we should remove setAsyncContext from RPCContext.